providers/mandrill/src/lib/mandrill.provider.ts

```diff
@@ -73,4 +75,55 @@ export class MandrillProvider implements IEmailProvider {
       code: CheckIntegrationResponseEnum.SUCCESS,
     };
   }
+
+  getMessageId(body: any | any[]): string[] {
+    if (Array.isArray(body)) {
+      return body.map((item) => item._id);
+    }
+
+    return [body._id];
+  }
+
+  parseEventBody(
+    body: any | any[],
+    identifier: string
+  ): IEmailEventBody | undefined {
+    if (Array.isArray(body)) {
+      body = body.find((item) => item._id === identifier);
+    }
+
+    if (!body) {
+      return undefined;
+    }
+
+    const status = this.getStatus(body.event);
+
+    if (status === undefined) {
+      return undefined;
+    }
+
+    return {
+      status: status,
+      date: new Date().toISOString(),
+      externalId: body._id,
+      attempts: body.attempt ? parseInt(body.attempt, 10) : 1,
+      response: body.response ? body.response : '',
+      row: body,
+    };
+  }
+
+  private getStatus(event: string): EmailEventStatusEnum | undefined {
+    switch (event) {
+      case 'open':
+        return EmailEventStatusEnum.OPENED;
+      case 'hard_bounce':
+        return EmailEventStatusEnum.BOUNCED;
+      case 'click':
+        return EmailEventStatusEnum.CLICKED;
+      case 'send':
+        return EmailEventStatusEnum.SENT;
+      case 'spam':
+        return EmailEventStatusEnum.SPAM;
```

Comment:
Creating a `MandrillStatusEnum` enum it would be the icing in the cake for this PR.

Reply:
For now I think it can safely live in `mandrill.provider.ts`.